source/ota_over_mqtt_demo.c

```diff
@@ -721,15 +785,82 @@ static void prvCommandCallback( void * pCommandContext,
 }
 
 /*-----------------------------------------------------------*/
+
+static void prvUnsubscribeCommandCallback( void * pxCommandContext,
+                                           MQTTAgentReturnInfo_t * pxReturnInfo )
+{
+    CommandContext_t * pxApplicationDefinedContext = ( CommandContext_t * ) pxCommandContext;
+    const char * pcTopicBuffer = ( const char * ) pxApplicationDefinedContext->pArgs;
+
+    /* Store the result in the application defined context so the task that
+     * initiated the subscribe can check the operation's status.  Also send the
+     * status as the notification value.  These things are just done for
+     * demonstration purposes. */
+    pxApplicationDefinedContext->xReturnStatus = pxReturnInfo->returnCode;
+
+    /* Check if the subscribe operation is a success. Only one topic is
+     * subscribed by this demo. */
+    if( pxReturnInfo->returnCode == MQTTSuccess )
+    {
+        /* Add subscription so that incoming publishes are routed to the application
+         * callback. */
+        removeSubscription( xGlobalSubscriptionList,
+                            pcTopicBuffer,
+                            ( uint16_t ) strlen( pcTopicBuffer ) );
+    }
+
+    xTaskNotify( pxApplicationDefinedContext->xTaskToNotify,
+                 ( uint32_t ) ( pxReturnInfo->returnCode ),
+                 eSetValueWithOverwrite );
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvSubscribeCommandCallback( void * pxCommandContext,
+                                         MQTTAgentReturnInfo_t * pxReturnInfo )
+{
+    CommandContext_t * pxApplicationDefinedContext = ( CommandContext_t * ) pxCommandContext;
+    SubscriptionInfo_t * pxSubscriptionInfo = ( SubscriptionInfo_t * ) pxApplicationDefinedContext->pArgs;
+    const char * pcTopicBuffer = pxSubscriptionInfo->pcTopicFilter;
+
+    /* Store the result in the application defined context so the task that
+     * initiated the subscribe can check the operation's status.  Also send the
+     * status as the notification value.  These things are just done for
+     * demonstration purposes. */
+    pxApplicationDefinedContext->xReturnStatus = pxReturnInfo->returnCode;
+
+    /* Check if the subscribe operation is a success. Only one topic at a time is
+     * subscribed by this demo. */
+    if( ( pxReturnInfo->returnCode == MQTTSuccess ) &&
+        ( pxReturnInfo->pSubackCodes[ 0 ] != MQTTSubAckFailure ) )
+    {
+        /* Add subscription so that incoming publishes are routed to the application
+         * callback. */
+        addSubscription( xGlobalSubscriptionList,
+                         pcTopicBuffer,
+                         ( uint16_t ) strlen( pcTopicBuffer ),
+                         pxSubscriptionInfo->pxIncomingPublishCallback,
+                         pxApplicationDefinedContext );
```

Comment:
This also used to be NULL before we made our API changes
```suggestion
                         NULL );
```